### DISCUSSION TITLE
How get all media in a related model

### DISCUSSION THREAD

**Discussion by Drummond-Dev** [spatie/laravel-medialibrary#2944] [Q&A]
I have 2 models, the first one of products and the second one of variations.
I created a component with Livewire that will export a main image of the product and all the images of the variations.

How can I get all the variation images? I tried.

$this->product->variations->getMedia('variations');

And returns that the method does not exist

**Comment by PaolaRuby** [comment]
Use nested relations

**Reply by Drummond-Dev** [reply]
What do you mean by nested relations? Can you give an example?

Thanks!

**Reply by Drummond-Dev** [reply]
Sorry I'm new to using the library

**Reply by PaolaRuby** [reply]
It is from laravel, not from this library
[eloquent-relationships#nested](https://laravel.com/docs/9.x/eloquent-relationships#nested-eager-loading)
Use it with a loop
